src/XrdCeph/XrdCephPosix.cc

```diff
@@ -706,8 +707,15 @@ int ceph_posix_open(XrdOucEnv* env, const char *pathname, int flags, mode_t mode
        logwrapper((char*)"Could not find size or stripe_unit xattr for %s", fr.name.c_str());
       }
      else{
-       fr.stripeUnit = std::stoull(d_stripeUnit.c_str());
-       fr.objectSize = std::stoull(d_objectSize.c_str());
+       char *cleanStripeUnit = new char[1024];
+       char *cleanObjectSize = new char[1024];
+       unsigned int stripeUnitLength = std::min((unsigned int)1023, d_stripeUnit.length());
+       unsigned int objectSizeLength = std::min((unsigned int)1023, d_objectSize.length());
+       strncpy( cleanStripeUnit, d_stripeUnit.c_str(), stripeUnitLength );
+       strncpy( cleanObjectSize, d_objectSize.c_str(), objectSizeLength );
```

Comment:
Are you sure that `strncpy` will add NULL-terminator? I think it will not happen, since `.c_str()` method of bufferlist do not add it. Manual addition of '\0' may be necessary.

Reply:
new would set all values to null by default. static alloc doesn't. changed accordingly